pkg/util/httputil/client.go

```diff
@@ -78,6 +98,37 @@ func (c *Client) Get(ctx context.Context, url string) (resp *http.Response, err
 	return c.Do(req)
 }
 
+// Put does like http.Client.Put but uses the provided context and obeys its cancellation.
```

Comment:
Nit: same as above. Also, the comment should mention `ContentType` and `Authorization`.